cartridges/int_adyen_SFRA/cartridge/controllers/Adyen.js

```diff
@@ -27,58 +27,59 @@ server.get('Adyen3D', server.middleware.https, function (req, res, next) {
 
 server.post('AuthorizeWithForm', server.middleware.https, function (req, res, next) {
   var adyen3DVerification = require('int_adyen_overlay/cartridge/scripts/adyen3DVerification');
-  var order = session.custom.order;
-  var paymentInstrument = session.custom.paymentInstrument;
-  if (session.custom.MD == req.form.MD) {
-    var result = adyen3DVerification.verify({
-      Order: order,
-      Amount: paymentInstrument.paymentTransaction.amount,
-      CurrentRequest: req.request,
-      MD: req.form.MD,
-      PaResponse: req.form.PaRes
-    });
 
-    // if error, return to checkout page
-    if (result.error || result.Decision != 'ACCEPT') {
-      Transaction.wrap(function () {
-        OrderMgr.failOrder(order);
+  if(session.custom.orderNo && session.custom.paymentMethod) {
+    var order = OrderMgr.getOrder(session.custom.orderNo);
+    var paymentInstrument = order.getPaymentInstruments(session.custom.paymentMethod)[0];
```

Comment:
I would suggest some defensive programming since it can happen that the `order.getPaymentInstruments(session.custom.paymentMethod)` returns undefined and the undefined[0] will return an error